src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,88 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.YamlDSLModelHelper;
+
+/**
+ * @author Lars Heinemann
+ */
+public class CamelKYamlDSLParser extends ParserFileHelper {
+	
+	public static final String STEP_KEY = "steps:";
+	public static final String URI_KEY = "uri:";
+	public static final String REST_KEY = "rest:";
+	public static final String FROM_KEY = "- from:";
+	public static final String TO_KEY = "- to:";
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		String camelComponentURI = null;
+		String trLine = line.trim();
+		if (trLine.startsWith(URI_KEY) && URI_KEY.length() < characterPosition) {
+			camelComponentURI = extractCamelCOmponentURI(trLine, URI_KEY);
+		} else if(trLine.startsWith(FROM_KEY) && trLine.indexOf('"') >= trLine.indexOf(':') + 1) {
+			camelComponentURI = extractCamelCOmponentURI(trLine, FROM_KEY);
+		} else if(trLine.startsWith(TO_KEY) && trLine.indexOf('"') >= trLine.indexOf(':') + 1) {
+			camelComponentURI = extractCamelCOmponentURI(trLine, TO_KEY);
+		}
+		return camelComponentURI;
+	}
+
+	@Override
+	public CamelURIInstance createCamelURIInstance(TextDocumentItem textDocumentItem, Position position,
+			String camelComponentUri) {
+		CamelURIInstance uriInstance = new CamelURIInstance(camelComponentUri, new YamlDSLModelHelper(getCorrespondingLine(textDocumentItem, position.getLine())), textDocumentItem);
+		int start = getStartCharacterInDocumentOnLinePosition(textDocumentItem, position);
+		uriInstance.setStartPositionInDocument(new Position(position.getLine(), start));
+		uriInstance.setEndPositionInDocument(new Position(position.getLine(), start+camelComponentUri.length()));
+		return uriInstance;
+	}
+
+	private String extractCamelCOmponentURI(String trLine, String key) {
+		return trLine.substring(trLine.indexOf(key)+ key.length()).trim().replaceAll("\"", "");
+	}
+
+	private int getStartCharacterInDocumentOnLinePosition(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		return line.indexOf('"') + 1;
+	}
+
+	@Override
+	public int getPositionInCamelURI(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		return position.getCharacter() - line.indexOf('"')-1;
+	}
+
+	public String getCorrespondingLine(TextDocumentItem textDocumentItem, int lineNumber) {
```

Comment:
it is returning the corresponding type, not the corresponding line

Reply:
added